string mode
int32 count
---
bool finish